openmmtools/multistate/multistatesampler.py

```diff
@@ -1711,6 +1725,35 @@ def _initialize_context_caches(self):
         self.energy_context_cache = cache.global_context_cache
         self.sampler_context_cache = cache.global_context_cache
 
+    def _update_timing(self, iteration_time, partial_total_time, run_initial_iteration, iteration_limit):
+        """
+        Function that computes and transmits timing information to reporter.
+
+        Parameters
+        ----------
+        iteration_time : float
+            Time took in the iteration.
+        partial_total_time : float
+            Partial total time elapsed.
+        run_initial_iteration : int
+            Iteration where to start/resume the simulation.
+        iteration_limit : int
+            Hard limit on number of iterations to be run by the sampler.
+        """
+        self._timing_data["iteration_seconds"] = iteration_time
+        self._timing_data["average_seconds_per_iteration"] = \
+            partial_total_time / (self._iteration - run_initial_iteration)
+        estimated_timedelta_remaining = datetime.timedelta(
+            seconds=self._timing_data["average_seconds_per_iteration"] * (iteration_limit - self._iteration)
+        )
+        estimated_finish_date = datetime.datetime.now() + estimated_timedelta_remaining
+        self._timing_data["estimated_time_remaining"] = str(estimated_timedelta_remaining)  # Putting it in dict as str
+        self._timing_data["estimated_iso_finish_date"] = estimated_finish_date.strftime("%Y-%b-%d-%H:%M:%S")
```

Comment:
The time zone is not specified. This should be something like `estimated_localtime_finish_date` if you're using local time.